### ISSUE TITLE
Colour by property - a way to opt out the spacial elements?

### ISSUE THREAD

**Issue by BIMvoice** [IfcOpenShell/IfcOpenShell#5041]
Hey,

What do you guys think about having a way to sort out the spatial elements? Most of the time I am using this to check the physical elements and it's annoying to have to remember to not consider the spatial ones.

Would there be any elegant way, simple to implement that could help with this?

![image](https://github.com/user-attachments/assets/d6eb25b0-c30d-4f4a-94d4-25b94c25f618)

**Comment by falken10vdl** [comment]
@BIMvoice Can you explain it again? I do not understand what you are after. Currently the color by Property shows the classes sorted alphabetically. What would you like?
Thanks!

**Comment by BIMvoice** [comment]
> [@BIMvoice](https://github.com/BIMvoice) Can you explain it again? I do not understand what you are after. Currently the color by Property shows the classes sorted alphabetically. What would you like? Thanks!

Hey @falken10vdl thank you for popping this up. I'm sorry, I uploaded another image than the one I had in mind.

What I try to say is that only IfcElement should show up in this list. What is the point in having the ones that I crossed off in the image? I think only the physical ones are relevant for this list.

Thank you for your help!

<img width="645" alt="Image" src="https://github.com/user-attachments/assets/fc7cb6bf-0d0d-4dee-a14b-0edee0f09ed5" />

**Comment by theoryshaw** [comment]
Maybe add a filter property that would filter the list down?
I could see some users wanting to color IfcSite or IfcSpace, etc. 
![Image](https://github.com/user-attachments/assets/8fdeba18-21d4-433e-93f4-f4dd8c29e0cc)

**Comment by falken10vdl** [comment]
The filter idea I think it is very elegant and could probably be reused in other cases. Something similar to the filtering in excel. Let me propose you somehing like for your review.
Thanks!

**Comment by Andrej730** [comment]
@BIMvoice 

I think there are two separate issues going on.

1. It's a bug that it shows axis placements and owner history there, can you please share example to reproduce the issue? I couldn't reproduce it.

![Image](https://github.com/user-attachments/assets/cd92d873-037f-4bc3-b418-8bd8daca670a)

2. Regarding the building storeys and etc. I think we should just filter all elements that don't have meshes (e.g. ignore objects with `obj.data is None`), so all those empties won't appear in this menu in the first place and there won't be need for a special filter for this.

If we want to add a filter, we should somehow connect it to existing search filters (e.g. at least add a field so search query can be used to filter elements, not just by using class but by using the entire syntax we have).

**Comment by falken10vdl** [comment]
@Andrej730 does it make senses tu use UI_UL_List as to get directly methods for searching/sorting/filtering? As you say if there is no graphical representation, probably the list shludl be reduced already.
@theoryshaw and @BIMvoice what would then filtering be used for? If you hide a line for an element that has graphical representation. What is the intended behaviour? That the element's color is kept with the color it had previously? That the element itself is not shown (ie like.if it was transparent)?
Thanks!

**Comment by Andrej730** [comment]
I think the main intention here was to clear UIList from elements that has nothing to do with it. So, maybe just hiding elements without representations could be enough for now.
Alternatively, as a simple enchancement, there could be also a simple string field to input query from selector syntax to reduce amount of affected elements.

**Comment by falken10vdl** [comment]
@Andrej730 . Something like this?

<img width="571" alt="Image" src="https://github.com/user-attachments/assets/7adc7574-50d2-4511-8c34-290519c24851" />

Regards,
Vicente.

**Comment by Andrej730** [comment]
> Something like this?

Yeah, we can also include `CURVE` there, as we do have them from time to time.

**Comment by falken10vdl** [comment]
https://github.com/IfcOpenShell/IfcOpenShell/pull/6657
